### PR TITLE
Dash Radio Rich Presence

### LINKED ISSUE #2167: Dash Radio | dashradio.com
**Prerequisites and essential questions**
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [X] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?**
Cole#0001

**What's the name of the service?**
https://dashradio.com/ | Dash Radio

**What should the Presence display?**
The presence would ideally display the logo, the name of the service (Dash Radio), the title of the current track - the artist name, and the channel name. (Very similarly or identically to how the SiriusXM presence is implemented)

**If possible, please provide a logo for the service (512x512 minimum)**
Dash Radio Logo
Dash Radio Logo White Large

### PR DESCRIPTION
**Pull request by misapprehending**
Resolves #2167

### CODE REVIEW

**Comment by misapprehending** [comment]
Resolves #2167

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload proof of it working.
Description should be the description of the website not of what the presence is.

**Comment by misapprehending** [comment]
https://streamable.com/3bb1ka

Would you mind elaborating on:
"Description should be the description of the website not of what the presence is."

**Comment by ririxichan** [comment]
Description must be description of website, not description like "discord rpc for ..."

File: websites/D/Dash Radio/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "author": {
+    "name": "apprehends",
+    "id": "202236922217758721"
+  },
+  "service": "Dash Radio",
+  "description": {
+    "en": "Dash Radio is a premium radio website without the ads. Available on over 50 platforms. (IN ORDER FOR THIS TO WORK, TEXT MUST BE STATIC NOT SCROLLING)"
+  },
```
Comment (Bas950) [inline websites/D/Dash Radio/dist/metadata.json:9]: Use their official description:"
100% Free Music. More Variety. Zero Commercials. The Best DJ's & Stations. No BS

File: websites/D/Dash Radio/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  "url": "dashradio.com",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/o4Ea2f9.png",
+  "thumbnail": "https://i.imgur.com/o4Ea2f9.png",
+  "color": "#55effa",
```
Comment (Bas950) [inline websites/D/Dash Radio/dist/metadata.json:14]: thumbnail should be a promotional banner or a screenshot of the website homepage.

File: websites/D/Dash Radio/presence.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    songArtist = document.querySelector(
+      "header.MuiAppBar-root > div.music-dataview-container > span.App-Player-Song-Artist-Text"
+    );
+    if(songArtist == null) {
+      songArtistS = "None";
+    } else if(songArtist != null) {
+      songArtistS = songArtist.innerText;
+    }
+
+    presenceData.smallImageKey = "logo";
```
Comment (Bas950) [inline websites/D/Dash Radio/presence.ts:34]: Small image key is used for playing paused etc. icons
You can use our official ones from: https://github.com/PreMiD/Presences/tree/master/.resources

File: websites/D/Dash Radio/dist/metadata.json (hunk 1 above)
Comment (Bas950) [reply websites/D/Dash Radio/dist/metadata.json:9]: Also remove the disclaimer. You should make it work either way, you can probs just read the elements or through something else?

**Review by Bas950 — CHANGES_REQUESTED** [review]
Add support for other parts of the website, since it might look broken otherwise.

File: websites/D/Dash Radio/dist/metadata.json (hunk 1 above)
Comment (misapprehending) [reply websites/D/Dash Radio/dist/metadata.json:9]: I attempted to make it work but it just didn't want to read the span element for some reason. It's just printing as null

File: websites/D/Dash Radio/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "service": "Dash Radio",
+  "description": {
+    "en": "100% Free Music. More Variety. Zero Commercials. The Best DJ's & Stations. No BS"
+  },
+  "url": "dashradio.com",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/o4Ea2f9.png",
+  "thumbnail": "https://i.imgur.com/8612XSi.png",
+  "color": "#55effa",
```
Comment (Bas950) [inline websites/D/Dash Radio/dist/metadata.json:14]: thumbnail should be a promotional banner or a screenshot of the website homepage.

**Review by Bas950 — CHANGES_REQUESTED** [review]
Upload new proof of it working

File: websites/D/Dash Radio/dist/metadata.json (hunk 4 above)
Comment (misapprehending) [reply websites/D/Dash Radio/dist/metadata.json]: > Upload new proof of it working

https://streamable.com/wqyoyn

**Review by Bas950 — APPROVED** [review]

**Comment by misapprehending** [comment]
I noticed you approved one of my other pull requests, do you need to add yourself to this one as well? @Timeraa

**Review by Timeraa — APPROVED** [review]

**Comment by Timeraa** [comment]
Nope

**Pull request merged**